frontend/src/concepts/k8s/NameDescriptionField.tsx

```diff
@@ -61,15 +69,24 @@ const NameDescriptionField: React.FC<NameDescriptionFieldProps> = ({
             id={nameFieldId}
             data-testid={nameFieldId}
             name={nameFieldId}
-            value={data.name}
-            onChange={(e, name) => setData({ ...data, name })}
-            maxLength={maxLength}
+            value={name}
+            onChange={(_e, value) => {
+              setData({ ...data, name: value });
+              onNameChange?.();
+            }}
+            onBlur={onNameBlur}
+            validated={hasMaxLengthError ? 'error' : 'default'}
           />
-          {maxLength && (
+
+          {hasMaxLengthError && (
             <HelperText>
-              <HelperTextItem>{`Cannot exceed ${maxLength} characters`}</HelperTextItem>
+              <HelperTextItem variant="error" hasIcon>
```

Comment:
keep this info for the same reason

Reply:
Design wanted to avoid having stacked helper text, which is why this was converted to an error. cc @yannnz